File: go/Gas Station.go
func canCompleteCircuit(gas []int, cost []int) int {
	n := len(gas)
	totalGas := 0
	currentGas := 0
	startIndex := 0

	for i := 0; i < n; i++ {
		totalGas += gas[i] - cost[i]
		currentGas += gas[i] - cost[i]

		if currentGas < 0 {
			currentGas = 0
			startIndex = i + 1
		}
	}

	if totalGas >= 0 {
		return startIndex
	}
	return -1
}

File: python3/Gas Station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        n = len(gas)
        total_gas = 0
        current_gas = 0
        start_index = 0

        for i in range(n):
            total_gas += gas[i] - cost[i]
            current_gas += gas[i] - cost[i]

            if current_gas < 0:
                current_gas = 0
                start_index = i + 1

        if total_gas >= 0:
            return start_index
        else:
            return -1